File: scripts/copy.py
import shutil
import brotli
import mmap
from pathlib import Path

BASE_DIR = Path(__file__).resolve().parent

DIST_DIR = Path(
    BASE_DIR.parent,
    "dist",
    "django-hstore-widget",
)

TARGET_DIR = Path(
    BASE_DIR.parent,
    "src",
    "django_hstore_widget",
    "static",
    "admin",
    "js",
    "django_hstore_widget",
)

if TARGET_DIR.exists() and TARGET_DIR.is_dir():
    shutil.rmtree(TARGET_DIR)

TARGET_DIR.mkdir(parents=True, exist_ok=True)

files = list(DIST_DIR.glob("*.js"))

if not files:
    raise Exception("There are no js files. Re-run `npm build`")

for js_file in files:
    if js_file.stat().st_size == 0:
        continue

    with js_file.open("rb") as f:
        with mmap.mmap(f.fileno(), 0, access=mmap.ACCESS_READ) as mm:
            compressed_data = brotli.compress(mm)

    compressed_file = TARGET_DIR / f"{js_file.name}.br"
    compressed_file.write_bytes(compressed_data)

print("DONE")

File: scripts/copy.ts
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

// Get current directory (ESM equivalent of __dirname)
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const BASE_DIR = path.resolve(__dirname);
const DIST_DIR = path.resolve(BASE_DIR, '..', 'dist', 'django-hstore-widget');
const TARGET_DIR = path.resolve(BASE_DIR, '..', 'src', 'django_hstore_widget', 'static', 'admin', 'js', 'django_hstore_widget');

try {
    // Remove target directory if exists
    if (fs.existsSync(TARGET_DIR)) {
        fs.rmSync(TARGET_DIR, { recursive: true, force: true });
    }

    // Create target directory
    fs.mkdirSync(TARGET_DIR, { recursive: true });

    // Find JS files in dist directory
    const files = fs
        .readdirSync(DIST_DIR)
        .filter(file => file.endsWith('.js'))
        .map(file => path.join(DIST_DIR, file));

    if (files.length === 0) {
        throw new Error('There are no js files. Re-Run `npm run build`');
    }

    // Copy files
    let copiedCount = 0;
    for (const file of files) {
        const stats = fs.statSync(file);
        if (stats.size > 0) {
            const targetFile = path.join(TARGET_DIR, path.basename(file));
            fs.copyFileSync(file, targetFile);
            copiedCount++;
        }
    }

    if (copiedCount > 0) {
        console.log('DONE');
    } else {
        console.warn('No non-empty files were copied');
    }
} catch (error) {
    console.error('Error:', (error as Error).message);
    process.exit(1);
}
